cmd/policy-assistant/pkg/kube/ikubernetes.go

```diff
@@ -91,9 +104,12 @@ type MockNamespace struct {
 }
 
 type MockKubernetes struct {
-	Namespaces map[string]*MockNamespace
-	passRate   float64
-	podID      int
+	AdminNetworkPolicies func() ([]v1alpha12.AdminNetworkPolicy, error)
```

Comment:
I did it this way because I needed a way to express more real-life scenarios. For example, the server doesn't return a response in the specified time. 

With the current design pattern, this is hard to recreate as it simply returns data ignoring all possible issues that can come from the communication layer. So if you are Ok with ignoring such scenarios/tests, let me know.

Reply:
We could make the MockKubernetes error out with a method call like k.ErrorNextTime()?